contributing/golang.md

```diff
@@ -0,0 +1,27 @@
+# Nomad Go Version
+
+Nomad attempts to build each Y release (where 0.9.5 means X=0, Y=9, Z=5) with
```

Comment:
You and I know that "attempts" here means that we're not making any promises, but with the context of "build" this sounds more like we're running matrix tests or something that we don't intend to imply. Maybe we could remove this ambiguity by saying something like:

> Nomad is intended to be built with a specific version of the Go toolchain for each release.
Generally, each Y release of Nomad (where 0.9.5 means X=0, Y=9, Z=5) will update to
the latest version of the Go toolchain available at the time.

Reply:
Ah good point, that is confusing. Stealing your wording as I clearly have not had enough coffee yet. :sweat_smile: